src/ReportManager/DetailsSection/SchemaForm/index.js

```diff
@@ -15,14 +17,51 @@ export const FIELDS = {
   [FORM_ELEMENT_TYPES.TEXT]: Text,
 };
 
-const SchemaForm = ({ formData, onFormChange, onFormSubmit, renderSubmitButton, schema }) => {
-  const fields = useMemo(() => schema ? makeFieldsFromSchema(schema) : {}, [schema]);
+const ajv = new Ajv2020({ allErrors: true });
+
+const SchemaForm = ({ formData, onFormDataChange, onFormSubmit, renderSubmitButton, schema }) => {
+  const { t } = useTranslation('reports', { keyPrefix: 'reportManager.detailsSection.schemaForm' });
+
+  const [fieldErrors, setFieldErrors] = useState({});
+
+  const fields = useMemo(() => makeFieldsFromSchema(schema), [schema]);
+
+  const fieldValues = useMemo(() => Object.entries(formData).reduce((accumulator, [fieldId, fieldValue]) => {
+    // TODO: Collections will require recusivity here.
+
+    return { ...accumulator, [fieldId]: fieldValue };
+  }, {}), [formData]);
+
+  const validate = useMemo(() => ajv.compile(schema.json), [schema.json]);
+
+  const onFieldChange = useCallback((fieldId, value) => {
+    // TODO: Collections will require recusivity here.
+    // If the value is empty, set it as undefined so AJV validation returns errors for required fields.
+    const newFormData = { ...formData, [fieldId]: value || undefined };
+
+    onFormDataChange(newFormData);
+    setFieldErrors((fieldErrors) => ({ ...fieldErrors, [fieldId]: undefined }));
+  }, [formData, onFormDataChange]);
 
   const onSubmit = (event) => {
     event.preventDefault();
 
-    // TODO: Add form validation with AJV before submission and show errors.
-    onFormSubmit({ formData });
+    const valid = validate(formData);
+    if (!valid) {
+      const fieldErrors = validate.errors.reduce((accumulator, error) => {
+        if (error.keyword === 'required') {
+          return { ...accumulator, [error.params.missingProperty]: t('errors.required') };
+        }
+
+        // TODO: Transform missing errors.
```

Comment:
Everytime the user tries to submit we validate the form data and transform the AJV error object into a flat object that maps a i18n translated error message by the field id. For now I only did the transformation of the only error supported by the text field, which is when the field is required and the user doesn't provide a value. We should keep adding transformations as we add the missing fields in the next PRs.